algorithms/polish.sbl

```diff
@@ -0,0 +1,200 @@
+      'a{l/}e{s'}'    'ia{l/}e{s'}'    'i{l/}e{s'}'    // past 2nd person singular masculine (czytałeś, leciałeś, chodziłeś)
+      'a{l/}a{s'}'    'ia{l/}a{s'}'    'i{l/}a{s'}'    // past 2nd person singular feminine (czytałaś, leciałaś, chodziłaś)
+      'a{l/}'         'ia{l/}'         'i{l/}'         // past 3rd person singular masculine (czytał, leciał, chodził)
+      'a{l/}a'        'ia{l/}a'        'i{l/}a'        // past 3rd person singular feminine (czytała, leciała, chodziła)
+      'a{l/}o'        'ia{l/}o'        'i{l/}o'        // past 3rd person singular neuter (czytało, leciało, chodziło)
+      'ali{s'}my'     'ieli{s'}my'     'ili{s'}my'     // past 1st person plural virile (czytaliśmy, lecieliśmy, chodziliśmy)
+      'a{l/}y{s'}my'  'ia{l/}y{s'}my'  'i{l/}y{s'}my'  // past 1st person plural nonvirile (czytałyśmy, leciałyśmy, chodziłyśmy)
+      'ali{s'}cie'    'ieli{s'}cie'    'ili{s'}cie'    // past 2nd person plural virile (czytaliście, lecieliście, chodziliście)
+      'a{l/}y{s'}cie' 'ia{l/}y{s'}cie' 'i{l/}y{s'}cie' // past 2nd person plural nonvirile (czytałyście, leciałyście, chodziłyście)
+      'ali'           'ieli'           'ili'           // past 3rd person plural virile (czytali, lecieli, chodzili)
+      'a{l/}y'        'ia{l/}y'        'i{l/}y'        // past 3rd person plural nonvirile (czytały, leciały, chodziły)
+      'aj'     // imperative 2nd person singular (czytaj)
+      'ajcie'  // imperative 2nd person plural (czytajcie)
+        (delete)
+    )
+  )
+
+  define short_verb as (
+    // There are short verbs whose root consists of only one consonant, e.g. być, żyć.
+    // Stemming them to one letter would merge them with these letters used in other
+    // contexts, which is undesirable. But let's at least merge all past tense forms
+    // together, e.g. byłem, byłam, byłyśmy, etc. to był.
+    ([substring]) among (
+      '{l/}e{s'}'
+      '{l/}a{s'}'
+      'li{s'}my'
+      '{l/}y{s'}my'
+      'li{s'}cie'
+      '{l/}y{s'}cie'
+        (<- '{l/}')
+      // Handle the imperative 2nd person plural form (chodźcie) here as well.
+      // We cannot do that earlier, because that would also affect the 2nd person
+      // plural past tense forms.
+      'cie'
+        (delete)
+    )
+  )
+
+  define verb2 as (
+    // -szą can be also part of comparative forms like piękniejszą,
+    // so we need to process it after we are done with adjectival forms.
+    setlimit tomark p0 for ([substring]) among (
+      'sz{ek}' // present 1st person singular (noszę)
+      'sz{ak}' // present 3rd person plural (noszą)
+        (<- 's')
+      '{ek}'   // present 1st person singular (lecę)
+        (delete)
+    )
+  )
+
+  define noun as (
+    setlimit tomark p1 for ([substring]) among (
+      'a' 'o'                   // singular nominative (książka, lato)
+      'i' 'y' 'u' 'ia'          // singular genitive (książki, kobiety, stołu, słonia)
```

Comment:
It looks like `'y'` here can't match because it's also handled in `adjectival`.  The action there is `delete` plus possible additional removals rather than just `delete`, but quite likely that's OK.